### PR TITLE
Add new configs

### PR DESCRIPTION
**Pull request by ashwinvaidya17**
## Description

Provide a summary of the modification as well as the issue that has been resolved.

- Adds new config

## Changes

<details>
<summary>Describe the changes you made</summary>

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactor (non-breaking change which refactors the code base)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

</details>

## Checklist

<details>
<summary>Ensure that you followed the following</summary>

- [ ] I have added a summary of my changes to the [CHANGELOG](https://github.com/openvinotoolkit/anomalib/blob/main/CHANGELOG.md) (not for minor changes, docs and tests).
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas (If applicable)
- [ ] I have made corresponding changes to the documentation (If applicable)
- [ ] I have added tests that prove my fix is effective or that my feature works (If applicable)

</details>

### CODE REVIEW

File: src/anomalib/models/cflow/anomaly_map.py [hunk 1]
```diff
@@ -18,12 +19,12 @@ class AnomalyMapGenerator(nn.Module):
...
         self,
         image_size: ListConfig | tuple,
-        pool_layers: list[str],
+        pool_layers: Sequence[str],
     ) -> None:
         super().__init__()
         self.distance = torch.nn.PairwiseDistance(p=2, keepdim=True)
         self.image_size = image_size if isinstance(image_size, tuple) else tuple(image_size)
-        self.pool_layers: list[str] = pool_layers
+        self.pool_layers: Sequence[str] = pool_layers
```
Comment (samet-akcay) [inline src/anomalib/models/cflow/anomaly_map.py:27]: Just curiosity, why is this changed to `Sequence`? CLI-related?

File: src/anomalib/models/cflow/lightning_model.py [hunk 2]
```diff
@@ -64,16 +64,9 @@ def __init__(
...
-            This method is used for the existing CLI.
-            When PL CLI is introduced, configure optimizers method will be
-                deprecated, and optimizers will be configured from either
-                config.yaml file or from CLI.
-
-        Returns:
+        Returns
         -------
-            Optimizer: Adam optimizer for each decoder
+                    Optimizer: Adam optimizer for each decoder
```
Comment (samet-akcay) [inline src/anomalib/models/cflow/lightning_model.py:69]:
```suggestion
    Optimizer: Adam optimizer for each decoder
```

File: src/anomalib/models/components/sampling/k_center_greedy.py [hunk 3]
```diff
@@ -6,9 +6,9 @@
 
 
 import torch
+from rich.progress import track
```
Comment (samet-akcay) [inline src/anomalib/models/components/sampling/k_center_greedy.py:9]: Would it be an idea to replace all of the tqdm with track? Maybe in another PR?

File: src/anomalib/models/csflow/lightning_model.py [hunk 4]
```diff
@@ -102,6 +101,21 @@ def trainer_arguments(self) -> dict[str, Any]:
         """CS-Flow-specific trainer arguments."""
         return {"gradient_clip_val": 1, "num_sanity_val_steps": 0}
 
+    def configure_optimizers(self) -> torch.optim.Optimizer:
+        """Configure optimizers.
+
+        Returns
+        -------
```
Comment (samet-akcay) [inline src/anomalib/models/csflow/lightning_model.py:108]:
```suggestion
        Returns:
```

File: src/anomalib/models/dfm/lightning_model.py [hunk 5]
```diff
@@ -24,9 +24,9 @@ class Dfm(AnomalyModule):
 
     Args:
     ----
+        input_size (tuple[int, int]): Input size for the model.
```
Comment (samet-akcay) [inline src/anomalib/models/dfm/lightning_model.py:27]: Should we also add a default value for the `input_size`? Is this specified in the paper?

File: src/anomalib/models/efficient_ad/lightning_model.py [hunk 6]
```diff
@@ -90,8 +89,8 @@ class EfficientAd(AnomalyModule):
 
     def __init__(
         self,
-        teacher_out_channels: int,
-        image_size: tuple[int, int],
+        input_size: tuple[int, int],
```
Comment (samet-akcay) [inline src/anomalib/models/efficient_ad/lightning_model.py:92]: Same as above, would it be an idea to provide a default value to `input_size`? Maybe we could discuss this?

File: src/configs/data/mvtec.yaml [hunk 7]
```diff
@@ -0,0 +1,18 @@
+class_path: anomalib.data.MVTec
```
Comment (samet-akcay) [inline src/configs/data/mvtec.yaml:1]: Not in this PR, but I think it is time to rename this to MVTecAD. This is the original name of this dataset, and we have multiple MVTec datasets now

File: src/configs/model/ai_vad.yaml
Comment (samet-akcay) [inline src/configs/model/ai_vad.yaml:1]: This config file does not have `metric` or `trainer` arguments. Is it because those are not needed?

Comment (samet-akcay) [reply src/configs/model/ai_vad.yaml:1]: I've noticed that some other models also do not have these sections

**Review by samet-akcay — CHANGES_REQUESTED** [review]
Thanks! The new config structure looks really clean!

I've just minor questions and comments

File: src/anomalib/models/cflow/anomaly_map.py (hunk 1 above)
Comment (ashwinvaidya17) [reply src/anomalib/models/cflow/anomaly_map.py:27]: I cannot assign default parameters if I keep the type as list. With Sequence, I can use both tuple and list

File: src/anomalib/models/dfm/lightning_model.py (hunk 5 above)
Comment (ashwinvaidya17) [reply src/anomalib/models/dfm/lightning_model.py:27]: We override the input size with that of the dataloader. And, if someone uses the API and sets the dataloader input size to a different value then it will throw an error.

Comment (samet-akcay) [reply src/anomalib/models/dfm/lightning_model.py:26]: yeah, they need to explicitly specify the size to avoid issues. But this is the current way to create the datamodule and the model via API, right?

```python
from anomalib.data import Visa
from anomalib.models import EfficientAd

datamodule = Visa(image_size=(256, 256), ...)
model = EfficientAd(input_size(256, 256))
...
```

File: src/anomalib/data/base/dataset.py [hunk 8]
```diff
@@ -172,11 +172,10 @@ def _setup(self) -> DataFrame:
...
-        |---|-------------------|-----------|-------------|------------------|-------|
-        |   | image_path        | label     | label_index | mask_path        | split |
-        |---|-------------------|-----------|-------------|------------------|-------|
-        | 0 | path/to/image.png | anomalous | 1           | path/to/mask.png | train |
-        |---|-------------------|-----------|-------------|------------------|-------|
+            |---|-------------------|-----------|-------------|------------------|-------|
+            |   | image_path        | label     | label_index | mask_path        | split |
+            |---|-------------------|-----------|-------------|------------------|-------|
+            | 0 | path/to/image.png | anomalous | 1           | path/to/mask.png | train |
+            |---|-------------------|-----------|-------------|------------------|-------|
```
Comment (samet-akcay) [inline src/anomalib/data/base/dataset.py:179]: I've noticed that this table is not properly shown on the documentation. We need to make this a bit html friendly. Of course, in another PR when we are working on the documentation :)

**Review by samet-akcay — APPROVED** [review]
lgtm, thanks!
